### PR TITLE
AVS 2021-01-01-preview

### PR DESCRIPTION
**Pull request by ctaggart**
Replacing #12762 to make it easier to review. We are wanting to go live by Monday, 2/15.

This is a new version 2021-01-01-preview for AVS (Azure VMware Solution).  It adds these operations:
- Datastores_Create
- Datastores_Delete
- Datastores_Get
- Datastores_List

Pleases see this [private pull request](https://github.com/Azure/azure-rest-api-specs-pr/pull/2691) for previous reviews.

There are no breaking changes in this API version. The CI labels are from breaking changes that were already approved for both Go & Python in https://github.com/Azure/azure-rest-api-specs/pull/12290. These Go [changes](https://github.com/AzureSDKAutomation/azure-sdk-for-go/pull/17107/files#diff-269e9e402a65d116290061cb2d8cadcc28710d38201be5332dab4de28d07a7c8R6-R16) are from #12290.

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [x] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? 2/15/2021
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/vmware/resource-manager/readme.java.md [hunk 1]
```diff
@@ -16,10 +16,24 @@ output-folder: $(azure-libraries-for-java-folder)/azure-mgmt-avs
...
 ```
 
+### Tag: package-2021-01-01-preview and java
+
+These settings apply only when `--tag=package-2021-01-01-preview -java` is specified on the command line.
+Please also specify the `--azure-libraries-for-java-folder=<path to the root directory of your azure-sdk-for-java clone>`.
+
+``` yaml $(tag) == 'package-2021-01-01-preview' && $(java) && $(multiapi)
+java:
+  namespace: com.microsoft.azure.management.avs.v2020_07_17_preview
```
Comment (ctaggart) [inline specification/vmware/resource-manager/readme.java.md:31]: needs to be updated to new API

File: specification/vmware/resource-manager/Microsoft.AVS/preview/2021-01-01-preview/examples/Datastores_Create.json [hunk 2]
```diff
@@ -0,0 +1,40 @@
+{
+  "parameters": {
+    "api-version": "2021-01-01-preview",
+    "subscriptionId": "{subscription-id}",
+    "resourceGroupName": "group1",
+    "privateCloudName": "cloud1",
+    "clusterName": "cluster1",
+    "datastoreName": "datastore1",
+    "datastore": {
```
Comment (ctaggart) [inline specification/vmware/resource-manager/Microsoft.AVS/preview/2021-01-01-preview/examples/Datastores_Create.json:9]: This is missing from the spec. I feel like the tooling should flag this as an error.

File: specification/vmware/resource-manager/Microsoft.AVS/preview/2021-01-01-preview/vmware.json [hunk 3]
```diff
@@ -4337,6 +4553,99 @@
...
+        "nfsProviderIp": {
+          "type": "string",
+          "description": "IP address of the NFS provider"
+        },
+        "nfsFilePath": {
+          "type": "string",
+          "description": "File path through which the NFS volume is exposed by the provider"
+        }
+      },
+      "x-ms-azure-resource": true
```
Comment (ctaggart) [inline specification/vmware/resource-manager/Microsoft.AVS/preview/2021-01-01-preview/vmware.json:4639]: This is not a resource . http://azure.github.io/autorest/extensions/#x-ms-azure-resource

File: specification/vmware/resource-manager/Microsoft.AVS/preview/2021-01-01-preview/vmware.json [hunk 4]
```diff
@@ -4337,6 +4553,99 @@
...
+          "items": {
+            "type": "string"
+          }
+        },
+        "lunName": {
+          "type": "string",
+          "description": "Name of the LUN to be used"
+        }
+      },
+      "x-ms-azure-resource": true
```
Comment (ctaggart) [inline specification/vmware/resource-manager/Microsoft.AVS/preview/2021-01-01-preview/vmware.json:4656]: This is not a resource.

File: specification/vmware/resource-manager/Microsoft.AVS/preview/2021-01-01-preview/vmware.json [hunk 5]
```diff
@@ -4337,6 +4553,99 @@
...
+          "description": "Resource ID of the Azure NetApp Files volume",
+          "$ref": "#/definitions/NetAppVolumeProperties"
+        },
+        "diskPoolVolume": {
+          "description": "Resource ID of the iSCSI target from a Storage Pool and LUN to be used",
+          "$ref": "#/definitions/DiskPoolVolumeProperties"
+        }
+      }
+    },
+    "NetAppVolumeProperties": {
```
Comment (ctaggart) [inline specification/vmware/resource-manager/Microsoft.AVS/preview/2021-01-01-preview/vmware.json:4627]: I don't think the "Properties" suffix is needed here. Generally, only used for direct children of resources.

File: specification/vmware/resource-manager/Microsoft.AVS/preview/2021-01-01-preview/vmware.json [hunk 6]
```diff
@@ -0,0 +1,5545 @@
...
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/privateCloudName"
+          },
+          {
+            "$ref": "#/parameters/clusterName"
+          },
+          {
+            "$ref": "#/parameters/datastoreName"
```
Comment (VamsiSangam) [inline specification/vmware/resource-manager/Microsoft.AVS/preview/2021-01-01-preview/vmware.json:1063]: Is this required even when we have the section below? ... Also does the same need to be done for GET/DELETE datastore resource as well?

File: specification/vmware/resource-manager/Microsoft.AVS/preview/2021-01-01-preview/vmware.json [hunk 7]
```diff
@@ -0,0 +1,5545 @@
...
+          },
+          {
+            "name": "datastore",
+            "description": "A datastore in a private cloud cluster",
+            "in": "body",
```
Comment (VamsiSangam) [inline specification/vmware/resource-manager/Microsoft.AVS/preview/2021-01-01-preview/vmware.json:1068]: Just for my knowledge... Is property used to to indicate if the input needs to be passed via request body, or in URL?

File: specification/vmware/resource-manager/Microsoft.AVS/preview/2021-01-01-preview/vmware.json (hunk 6 above)
Comment (ctaggart) [reply specification/vmware/resource-manager/Microsoft.AVS/preview/2021-01-01-preview/vmware.json:1063]: Yes, `datastoreName` is required for `Datastores_Creates`, `Datastores_Get`, & `Datastores_Delete`.  `Datastores_Create` is the only one that requires a body.

File: specification/vmware/resource-manager/Microsoft.AVS/preview/2021-01-01-preview/vmware.json (hunk 7 above)
Comment (ctaggart) [reply specification/vmware/resource-manager/Microsoft.AVS/preview/2021-01-01-preview/vmware.json:1068]: Yes, see https://swagger.io/docs/specification/2-0/describing-parameters/

File: specification/vmware/resource-manager/Microsoft.AVS/preview/2021-01-01-preview/examples/Datastores_Create.json [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+      }
+    }
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "id": "/subscriptions/{subscription-id}/resourceGroups/group1/providers/Microsoft.AVS/privateClouds/cloud1/clusters/cluster1/datastores/datastore1",
+        "name": "datastore1",
+        "properties": {
+          "provisioningState": "Succeeded"
```
Comment (varunkumta) [inline specification/vmware/resource-manager/Microsoft.AVS/preview/2021-01-01-preview/examples/Datastores_Create.json:24]: the responses should include the properties that were set in the request (exception is for secrets - which these are not)

File: specification/vmware/resource-manager/Microsoft.AVS/preview/2021-01-01-preview/examples/WorkloadNetworks_DeleteSegments.json [hunk 9]
```diff
@@ -0,0 +1,14 @@
+{
+  "parameters": {
+    "api-version": "2021-01-01",
```
Comment (varunkumta) [inline specification/vmware/resource-manager/Microsoft.AVS/preview/2021-01-01-preview/examples/WorkloadNetworks_DeleteSegments.json:3]: >2021-01-01 [](start = 20, length = 10)

these need to be 2021-01-01-preview

File: specification/vmware/resource-manager/Microsoft.AVS/preview/2021-01-01-preview/examples/Datastores_Create.json (hunk 8 above)
Comment (VamsiSangam) [reply specification/vmware/resource-manager/Microsoft.AVS/preview/2021-01-01-preview/examples/Datastores_Create.json:24]: My bad, cameron, could you plz match this with the properties in [Get](https://github.com/Azure/azure-rest-api-specs/blob/7de2664a252a8b629b85c115168ca4012ed70025/specification/vmware/resource-manager/Microsoft.AVS/preview/2021-01-01-preview/examples/Datastores_Get.json#L15)?

Comment (ctaggart) [reply specification/vmware/resource-manager/Microsoft.AVS/preview/2021-01-01-preview/examples/Datastores_Create.json:28]: fixed in 7781bc5

**Comment by ctaggart** [comment]
Thanks for the review yesterday @varunkumta. I made the requested ARM changes to the examples.

**Comment by ctaggart** [comment]
Thank for the ARM approval @varunkumta. Do you or @zhenglaizhang know who this should be assigned to in order to get merged?

**Comment by zhenglaizhang** [comment]
Hi @ctaggart I just assigned the PR to myself. And you need get review approval from the requested reviewers,

**Comment by ctaggart** [comment]
@zhenglaizhang, who do I need to get approval from? It has been approved by ARM.

**Review by VamsiSangam — APPROVED** [review]

**Review by cjain85 — APPROVED** [review]

**Review by Raaghavi — APPROVED** [review]

**Pull request merged**